### PR TITLE
Control cert-manager installation with a separate config value

### LINKED ISSUE #971: Helm chart: enabling webhook causes an outdated version of cert-manager to be installed.
Setting webhook.enabled to true causes deployment of cert-manager sub chart, at version 1.5.1. I already have cert-manager 1.14.1 installed in my cluster.

I was unable to find a way to deploy jenkins-operator using helm with webhook enabled that would not also deploy cert-manager.

I think that introducing a separate `cert-manager.enabled` configuration value and using it as the subchart condition would help.

We could also upgrade cert-manager dependency to a more recent version, when we're at it.

### PR DESCRIPTION
**Pull request by rkrzewski**
Fixes #971

### CODE REVIEW

File: chart/jenkins-operator/README.md
Comment (rkrzewski) [inline chart/jenkins-operator/README.md:1]: I've regenerated the file with helm-docs and some earlier changes were picked up

**Comment by rkrzewski** [comment]
I'd appreciate a review, @brokenpip3

**Comment by brokenpip3** [comment]
hei @rkrzewski thanks a lot for the PR, I will do a proper review asap.

In the mean time I saw that you bumped the certmanager dependency, however the previous maintainers have decided to add the certmanager crds into the crds folder in the chart: https://github.com/jenkinsci/kubernetes-operator/blob/master/chart/jenkins-operator/crds/cert-manager.crds.yaml 

I tried previously to fix that by using some helm hooks but then I was forced to work on more urgent stuff and I planned to take a look on that after the 0.9.0 (which I will finally restart to handle in these days): https://github.com/jenkinsci/kubernetes-operator/pull/810 

If you want to take a look it will be great, otherwise even if we put your new condition to off the helm chart will try to deploy the old certamanager crds anyway

**Comment by rkrzewski** [comment]
> If you want to take a look it will be great, otherwise even if we put your new condition to off the helm chart will try to deploy the old certamanager crds anyway

Sure I'll take look at it. @brokenpip3, do you think if the CRDs can be just wrapped in a conditional clause based on `cert-manager.enabled` value? Would that work? 

@sbandaru08 if we sort out the above, it'll fix the problem you describe.

**Comment by sbandaru08** [comment]
Sure thank you

**Comment by brokenpip3** [comment]
> Sure I'll take look at it. @brokenpip3, do you think if the CRDs can be just wrapped in a conditional clause based on `cert-manager.enabled` value? Would that work?

yes, but be aware that you can't have a go template inside the `crds` directory in helm, so you can't add guards, that's why I was trying with the helm hook

**Comment by rkrzewski** [comment]
@brokenpip3 oh, I didn't know that. I've read up on https://helm.sh/docs/chart_best_practices/custom_resource_definitions/ and maybe the "Method 2" would work? We could place the cert-manager CRDs into a separate sub-chart inside of jenkins-operator chart that use the same guard condition as cert-manager chart.

**Comment by brokenpip3** [comment]
> @brokenpip3 oh, I didn't know that. I've read up on https://helm.sh/docs/chart_best_practices/custom_resource_definitions/ and maybe the "Method 2" would work? We could place the cert-manager CRDs into a separate sub-chart inside of jenkins-operator chart that use the same guard condition as cert-manager chart.

Indeed could be a good tentative :)

**Comment by rkrzewski** [comment]
I've pushed [e2e32df](https://github.com/rkrzewski/jenkins-kubernetes-operator/commit/e2e32df1129e7e8b7d57a2afbd79ccaa9d83b673) to branch [rkrzewski:cert-manager](https://github.com/rkrzewski/jenkins-kubernetes-operator/tree/cert-manager) but it did not show up in the PR. Am I doing something wrong, or is github malfunctioning?

**Comment by rkrzewski** [comment]
@brokenpip3 I've moved the CRDs to separate subchart as discussed and it seems to work as expected:
```
helm template --include-crds --set cert-manager.enabled=false jenkins-operator .
```
generates manifests that include Jenkins CRD but none of cert-manager CRDs.

**Comment by brokenpip3** [comment]
This is great, ty. I will try the time to review this PR during this weekend

**Comment by sbandaru08** [comment]
@brokenpip3  Just wanted to know to see when can this change live ?

**Comment by brokenpip3** [comment]
> @brokenpip3 Just wanted to know to see when can this change live ?

I need to add some helm tests to be sure that this change will not destroy the current upgrade/install flow from both with/without validation webhook. It could take a couple of days or a couple of weeks, will depend on my free time

**Comment by rkrzewski** [comment]
> I need to add some helm tests
If this is something I could help you with, please let me know.

**Review by brokenpip3 — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**